lib/_stream_readable.js

```diff
@@ -72,7 +72,7 @@ function ReadableState(options, stream) {
   this.highWaterMark = (hwm || hwm === 0) ? hwm : defaultHwm;
 
   // cast to ints.
-  this.highWaterMark = ~~this.highWaterMark;
+  this.highWaterMark = Math.floor(this.highWaterMark);
```

Comment:
@thefourtheye what do you suggest? Can you make an example?
This was just changed at @mscdex suggestion from `Math.trunc`.

All negative values are interpreted as 0 by the machinery, so nothing should change.